### DISCUSSION TITLE
Tools and CI Meeting - Date 7/1 & 7/8

### DISCUSSION THREAD

**Discussion by makubacki** [tianocore/edk2#5843] [General]
# Status

Scheduled for 7/1 @ 4:30pm Pacific Time
Scheduled for 7/8 @ 4:30pm Pacific Time

# Agenda

1. Follow up on "Enabling the PR process for edk2-platforms and other TianoCore repositories.".
   - Get updates on action items from last week:
     - @mdkinney: Run an experiment with a member of `edk2-reviewers` (does not have write permission) in the Code Owners file and see if they are added to the PR without the ability to add push label
     - @bcran: Check with maintainers in edk2-platforms if they are okay with switching to Code Owners
       - Maintainers indicated they are happy to move to Pull Requests
       - Still need to follow on Code Owners
2. @bcran: Binary archive for C compiled base tools. What happened to https://github.com/tianocore/edk2-basetools?
   - Defer to email or next week when @mdkinney is available
3. Determine next steps for non-ASCII characters in a contributor's name for copyright.
   - See https://github.com/tianocore/edk2/pull/5840#issuecomment-2199914113
   - @mdkinney needed for feedback

Note: Agenda items could not move forward since @mdkinney was not in attendance today.

# Todos From Previous Meeting

- Agenda item (1)
- Follow up on `impact:security` vs CVE with infosec group.
- BZ to GitHub issue transition for new issues.
  - Next step: Send RFC with the idea and details about how issues would be processed and handled
- Investigate GitHub merge queue as a replacement for Mergify.

# Groups.io meeting invite:

https://edk2.groups.io/g/devel/viewevent?repeatid=43269&eventid=2159616&calstart=2024-01-08

# Microsoft Teams meeting

Join on your computer or mobile app
[Click here to join the meeting](https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2223af6561-6e1c-450d-b917-d9d674eb3cb6%22%7d)
Or call in (audio only)
+1 425-616-0754,,330776518# United States, Seattle
Phone Conference ID: 330 776 518#
Find a local number | [Reset PIN](https://mysettings.lync.com/pstnconferencing)
[Learn More](https://aka.ms/JoinTeamsMeeting) | [Meeting options](https://teams.microsoft.com/meetingOptions/?organizerId=23af6561-6e1c-450d-b917-d9d674eb3cb6&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&threadId=19_meeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw@thread.v2&messageId=0&language=en-US)

**Comment by spbrogan** [comment]
Updated this to include 7/8 as most of the agenda is the same as last week.  I hope anyone with questions or concerns can join.  Thanks

**Reply by Ashwin-Veeraiah** [reply]
Commenting here for the 7/15 meeting agenda incase it does not go up. Not able to add reviewers to PR to basetools. Checks passed

**Comment by spbrogan** [comment]
Add topic: Discuss labels and how to help clarify their intention.  
https://edk2.groups.io/g/discuss/message/1491

**Comment by spbrogan** [comment]
Brief notes:
1. @makubacki  and @spbrogan Add code owners to auto assign maintainers.  Create current codeowners file.  Move reviewers into code owners or other file as comments to support maintainers manually adding reviewers.  Evaluate how to verify codeowners is complete and matches expected behavior.

2. @makubacki  and @spbrogan Evaluate branch policy to require code owners to approve and switch to github merge queue (no more push label or mergify).
3. @bcran  drive discussion and enablement of edk2-platforms adoption of codeowners with same plan as above.
4. non-ascii characters in copyright comments.  All agree it would be nice, but belief is that some tools like iasl may still fail to parse files.  @mdkinney to discuss if there are any next steps.  
5. @makubacki to update github label descriptions and process.
6. @makubacki to send RFC for github issue enablement for new issues and step 2 bz transition.
7.  PR Queue length:
    A.  Consolidating CI on github actions would be ideal.  
    B.  Some optimizations could be done on ordering of ci steps, failing fast, etc to minimize PR queue time (@bcran to open issue on edk2-pytools). 
    C. @bcran offered to self-host agents on machines with ~300 cores to help with long queue times. 
    D.  All agreed this is not top priority at the moment.
